src/state/database.rs

```diff
@@ -231,4 +237,67 @@ impl DatabaseExt for Database {
 
         Ok(())
     }
+
+    async fn get_deposit_claim_txhash(
+        &self,
+        session: &mut ClientSession,
+        btc_txid: String,
+    ) -> Result<String, DatabaseError> {
+        let pipeline = vec![
+            doc! {
+                "$match": {
+                    "tx_id": btc_txid
+                }
+            },
+            doc! {
+                "$lookup": {
+                    "from": "deposit_claim_txs",
+                    "let": { "vout_value": "$vout", "tx_id": "$tx_id" },
+                    "pipeline": [
+                        {
+                            "$match": {
+                                "$expr": {
+                                    "$and": [
+                                        { "$eq": ["$identifier", { "$concat": ["$$tx_id", ":", { "$toString": "$$vout_value" }] }] },
```

Comment:
Can't we do the concat directly in rust instead of doing it on the db?

Reply:
I can't because I don't have the `vout` of the transaction that I need to get the `identifier` of the entry in `deposit_claim_txs` collection. 
Other option is to update the `deposit_claim_txs` indexer and add `tx_id` as we have it, so we don't have to concat the value in our query. If so I could directly query the indexer data rather than `claimed_runes_deposits` and then `deposit_claim_txs`. What do you think ?

Reply:
Then no worries, let's keep it simple, we will optimize if we notice it is too slow